clang/test/SemaSYCL/unique_stable_name.cpp

```diff
@@ -222,3 +199,15 @@ void use() {
   // expected-note@+1{{in instantiation of}}
   f2<St>();
 }
+
+// A previous implementation resulted in this being an example of the
+// kernel-ordering and lexical lambda ordering issue.
+void out_of_order_use() {
```

Comment:
We might also want to incorporate this test case into https://github.com/intel/llvm/blob/sycl/sycl/test/regression/unnamed-lambda.cpp#L1, to have a test closer to a real-life setup

Reply:
@AlexeySachkov : Just added the  test you had in your fork, hopefully that is sufficient for you.